### PR TITLE
Upgrade react

### PR DESCRIPTION
**Pull request by zacharyfmarion**
Upgrade react to v0.39 (the bastards are already on like 0.42). Was a pain in the ass so maybe we should consider freezing the version unless something necessary / critical changes. Once this PR is approved I will submit the android PR as they are dependent on the same react version.

### CODE REVIEW

File: iOS/chronreact/AppDelegate.m [hunk 1]
```diff
@@ -18,51 +19,31 @@ @implementation AppDelegate
 - (BOOL)application:(UIApplication *)application didFinishLaunchingWithOptions:(NSDictionary *)launchOptions
 {
   [BatchPush enableAutomaticDeeplinkHandling:NO];
+  
+  // Print out the Batch installation id for testing
```
Comment (zacharyfmarion) [inline iOS/chronreact/AppDelegate.m:23]: This is for testing purposes - logs out the batch id of the specific app instance. Can use this for notification testing over the Batch API. I can give you the file I used to test (just remind me).

File: index.ios.js [hunk 2]
```diff
@@ -9,21 +9,22 @@ import {
   TabBarIOS,
 } from 'react-native';
 
-const _ = require('underscore');
```
Comment (zacharyfmarion) [inline index.ios.js:12]: Changed imports to ES6 in this file

File: index.ios.js [hunk 3]
```diff
@@ -59,9 +60,6 @@ const styles = StyleSheet.create({
 });
 
 const chronreact = React.createClass({
-  propTypes: {
```
Comment (zacharyfmarion) [inline index.ios.js:62]: This was unnecessary.

File: src/components/PostListRow.js [hunk 4]
```diff
@@ -29,6 +29,7 @@ const styles = StyleSheet.create({
     borderColor: '#DDDDDD',
   },
   highlight: {
+    flex: 1,
```
Comment (zacharyfmarion) [inline src/components/PostListRow.js:32]: For some reason the listview was crashing on the new version without this.

File: iOS/chronreact/AppDelegate.m (hunk 1 above)
Comment (blakejkaplan) [reply iOS/chronreact/AppDelegate.m:23]: Sounds good. Did you resolve the double notification issue?

Comment (zacharyfmarion) [reply iOS/chronreact/AppDelegate.m:23]: No still figuring out what it is, I just upgraded the react version and got rid of redundant code.

**Comment by blakejkaplan** [comment]
screen shot 2017-03-05 at 3 26 09 pm

There are some visual issues.

**Comment by blakejkaplan** [comment]
screen shot 2017-03-05 at 3 31 21 pm

**Comment by blakejkaplan** [comment]
My weird formatting was due to outdated dependencies. Fixed by reinstalling all dependencies. PR looks good.

**Review by blakejkaplan — APPROVED** [review]
Looks good.

**Pull request merged**